test/scene-system/sdk/usingFetchAndWebsocket.spec.ts

```diff
@@ -0,0 +1,107 @@
+import * as sinon from 'sinon'
+import { createFetch, FetchFunction } from '../../../packages/scene-system/sdk/Fetch'
+import { createWebSocket } from '../../../packages/scene-system/sdk/WebSocket'
+
+const originalFetch: FetchFunction = async (resource: RequestInfo, init?: RequestInit) => {
+  return new Response()
+}
+
+class FakeWebSocket {
+  constructor(url: string | URL, protocols?: string | string[]) {
+
+  }
+}
+
+let originalWebSocket;
+
+before(() => {
+  originalWebSocket = WebSocket
+  // @ts-ignore
+  globalThis.WebSocket = FakeWebSocket
+})
+
+after(() => {
+  globalThis.WebSocket = originalWebSocket
+  originalWebSocket = null
+})
+
+describe('Safe fetch and WebSocket permissions', () => {
+  it('fetch', async () => {
```

Comment:
its a good practice to split the tests so when we break something its easier to debug.
```ts
describe('Wrapped websocket', () => {
  it('should log an error if the ws is not secure', () => {
  
  })

  it('should run successfully if the ws is secure', () => {
  })
})
```
```ts
describe('Wrapped fetch', () => {
  it(...)
})

etc
```